orangecontrib/text/nyt.py

```diff
@@ -73,21 +53,20 @@ class NYT:
 
     def __init__(self, api_key):
         self._api_key = api_key.strip()
-        self._query_url = ""
-        self.query_key = ""
+        self._query_url = None
+        self.query_key = None
+        self.includes_fields = None
 
-        self.cache_pth = os.path.join(environ.buffer_dir, "nytcache")
+        self.cache_path = None
+        cache_folder = os.path.join(environ.buffer_dir, "nytcache")
         try:
-            os.makedirs(self.cache_pth)
+            if not os.path.exists(cache_folder):
+                os.makedirs(cache_folder)
+            self.cache_path = os.path.join(cache_folder, "query_cache")
         except:
-            pass
-        self.query_cache = None
-        self.includes_fields = [False]*5
-
-    def set_query_url(self,
-                      query,
-                      year_from=None, year_to=None,
-                      text_includes=None):
+            print("WARNING: Could not assemble NYT query cache path.")
```

Comment:
Instead of printing the warning to stdout it is beter to use the `warning` module. For example:

    import warnings
    warnings.warn('Could not assemble NYT query cache path', RuntimeWarning)